diff --git a/drivers/input/touchscreen/s3c2410_ts.c b/drivers/input/touchscreen/s3c2410_ts.c
index 5a31c4f..2b1d623 100644
--- a/drivers/input/touchscreen/s3c2410_ts.c
+++ b/drivers/input/touchscreen/s3c2410_ts.c
@@ -64,6 +64,36 @@
 
 #include <linux/ts_filter.h>
 
+/* touchscreen configuration hack */
+
+#include "ts_filter_median.c"
+#include "ts_filter_mean.c"
+
+static struct ts_filter_median_configuration gta02_ts_median_config = {
+	.extent = 31,
+	.decimation_below = 5,
+	.decimation_threshold = 8 * 3,
+	.decimation_above = 10, /* when moving fast */
+};
+
+static struct ts_filter_mean_configuration gta02_ts_mean_config = {
+	.bits_filter_length = 3,
+	.averaging_threshold = 6 * 3,
+};
+
+static struct s3c2410_ts_mach_info gta02_ts_cfg = {
+	.delay = 10000,
+	.presc = 0xff, /* slow as we can go */
+	.filter_sequence = {
+		[0] = &ts_filter_median_api,
+		[1] = &ts_filter_mean_api,
+	},
+	.filter_config = {
+		[0] = &gta02_ts_median_config,
+		[1] = &gta02_ts_mean_config,
+	},
+};
+
 /* For ts.dev.id.version */
 #define S3C2410TSVERSION	0x0101
 
@@ -440,8 +470,8 @@ static int __init s3c2410ts_probe(struct platform_device *pdev)
 
 	/* create the filter chain set up for the 2 coordinates we produce */
 	ret = ts_filter_create_chain(
-		(struct ts_filter_api **)&info->filter_sequence,
-		(void *)&info->filter_config, ts.tsf, ARRAY_SIZE(ts.coords));
+		(struct ts_filter_api **)&gta02_ts_cfg.filter_sequence,
+		(void *)&gta02_ts_cfg.filter_config, ts.tsf, ARRAY_SIZE(ts.coords));
 	if (ret)
 		dev_info(&pdev->dev, "%d filter(s) initialized\n", ret);
 	else /* this is OK, just means there won't be any filtering */
